sdk/nodejs/index.ts

```diff
@@ -0,0 +1,489 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as bcryptjs from "bcryptjs";
+import * as crypto from "crypto";
+import * as fs from "fs";
+import * as os from "os";
+import * as Path from "path";
+import { sprintf } from "sprintf-js";
+import { v4 } from "uuid";
+import { gzipSync } from "zlib";
+
+const titleCase = require("title-case");
+
+/**
+ * abs returns the absolute value of a given number. See also the signum function.
+ *
+ * Examples:
+ *     - `abs(1)` returns `1`
+ *     - `abs(-1)` returns `1`
+ *     - `abs(-3.14)` returns `3.14`
+ *
+ * @param input number for which to return the absolute value
+ */
+export function abs(input: number): number {
+    return Math.abs(input);
+}
+
+/**
+ * basename returns the last element of a path.
+ *
+ * @param path the full path
+ */
+export function basename(path: string): string {
+    return Path.win32.basename(path);
+}
+
+/**
+ * base64decode decodes base64-encoded data and returns the utf-8-encoded string. An
+ * error is thrown if the argument is not valid base64 data.
+ *
+ * @param encoded valid base64-encoded string
+ */
+export function base64decode(encoded: string): string {
+    const decoded = new Buffer(encoded, "base64").toString("utf-8");
+    if (decoded.length !== ((encoded.length / 4) * 3)) {
+        throw new Error("failed to decode base64 data");
+    }
+
+    return decoded;
+}
+
+/**
+ * base64encode returns a base64-encoded version of the given string.
+ *
+ * @param unencoded the string to encode
+ */
+export function base64encode(unencoded: string): string {
+    return new Buffer(unencoded).toString("base64");
```

Comment:
I believe `new Buffer` is deprecated in Node 10.  We should probably use `Buffer.from`.

https://nodesource.com/blog/understanding-the-buffer-deprecation-in-node-js-10/

Reply:
:+1: